presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -764,17 +764,21 @@ public int read(byte[] buffer, int offset, int length)
                 int bytesRead = retry()
                         .maxAttempts(maxAttempts)
                         .exponentialBackoff(new Duration(1, TimeUnit.SECONDS), maxBackoffTime, maxRetryTime, 2.0)
-                        .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class)
+                        .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class, AbortedException.class)
                         .onRetry(STATS::newReadRetry)
                         .run("readStream", () -> {
                             seekStream();
                             try {
                                 return in.read(buffer, offset, length);
                             }
-                            catch (Exception e) {
-                                STATS.newReadError(e);
+                            catch (AbortedException abortedException) {
```

Comment:
`AbortedException` is thrown from many sdk methods if the thread making the call is interrupted, which may be true for limit queries or when a query gets canceled (I believe) etc. So I am not sure whether it's really an error condition, that's why I added this new catch block to swallow it, otherwise it was being propagated all the way up and I was seeing `Error processing split ...` messages around (also I wanted to get it out of the stats we collect as I see a lot of aborted exceptions in the stats that don't say much). If the exception is swallowed I had to return the number of bytes read and I think `0` is a suitable value (EOF is `-1`).

I closed the stream to be consistent with the other exception handlers.

Once we decide on the final solution I will apply it to `seek` too.